src/main/PDF/PdfControllerHelper.java

```diff
@@ -0,0 +1,292 @@
+        // If the field has children
+        List<PDField> childrenFields = ((PDNonTerminalField) field).getChildren();
+        fields.addAll(childrenFields);
+      } else {
+        if (field instanceof PDButton) {
+          if (field instanceof PDCheckBox) {
+            fieldsJSON.add(getCheckBox((PDCheckBox) field));
+          } else if (field instanceof PDPushButton) {
+            fieldsJSON.add(getPushButton((PDPushButton) field));
+          } else if (field instanceof PDRadioButton) {
+            fieldsJSON.add(getRadioButton((PDRadioButton) field));
+          }
+        } else if (field instanceof PDVariableText) {
+          if (field instanceof PDChoice) {
+            fieldsJSON.add(getChoiceField((PDChoice) field));
+          } else if (field instanceof PDTextField) {
+            fieldsJSON.add(getTextField((PDTextField) field));
+          }
+        } else if (field instanceof PDSignatureField) {
+          // Do nothing, as signatures are dealt with in findSignatureFields
+        }
+      }
+
+      // Delete field just gotten so we do not infinite recurse
+      fields.remove(0);
+    }
+    return fieldsJSON;
+  }
+
+  public static JSONObject getTextField(PDTextField field) {
+    String fieldName = field.getFullyQualifiedName();
+    String fieldType = "TextField";
+    String fieldValueOptions = "[]";
+    int numLines = DEFAULT_TEXT_FIELD_NUM_LINES;
+    String fieldQuestion = "Please Enter Your " + field.getPartialName();
+    return createFieldJSONEntry(fieldName, fieldType, fieldValueOptions, numLines, fieldQuestion);
+  }
+
+  public static JSONObject getCheckBox(PDCheckBox field) {
+    String fieldName = field.getFullyQualifiedName();
+    String fieldType = "CheckBox";
+    JSONArray optionsJSONArray = new JSONArray();
+    optionsJSONArray.put(field.getOnValue());
+    String fieldValueOptions = optionsJSONArray.toString();
+    int numLines = DEFAULT_CHECK_BOX_NUM_LINES;
+    String fieldQuestion = "Please select an option for " + field.getPartialName();
+    return createFieldJSONEntry(fieldName, fieldType, fieldValueOptions, numLines, fieldQuestion);
+  }
+
+  public static JSONObject getPushButton(PDPushButton field) {
+    String fieldName = field.getFullyQualifiedName();
+    String fieldType = "PushButton";
+    String fieldValueOptions = "[]";
+    int numLines = DEFAULT_PUSH_BUTTON_NUM_LINES;
+    String fieldQuestion = "Select the Button If You Want To " + field.getPartialName();
+    return createFieldJSONEntry(fieldName, fieldType, fieldValueOptions, numLines, fieldQuestion);
+  }
+
+  public static JSONObject getRadioButton(PDRadioButton field) {
+    String fieldName = field.getFullyQualifiedName();
+    String fieldType = "RadioButton";
+    JSONArray optionsJSONArray = new JSONArray();
+    for (String choice : field.getOnValues()) {
+      optionsJSONArray.put(choice);
+    }
+    String fieldValueOptions = optionsJSONArray.toString();
+    int numLines = 2 + optionsJSONArray.length();
```

Comment:
why are we adding 2 to the length?

Reply:
Because, we are calculating the number of "lines" a frontend feature takes up, and each additional option takes up 2 lines.